crates/ruff_python_formatter/src/lib.rs

```diff
@@ -120,7 +120,7 @@ impl From<ParseError> for FormatModuleError {
     }
 }
 
-#[tracing::instrument(level=Level::TRACE, skip_all, err)]
```

Comment:
This was causing tracing to print the error, but we want to handle reporting it downstream of this.

Reply:
Shouldn't it only print the error when the tracing level is `err`? @konstin can we tune our tracing setup in the cli instead? I would want to keep errors showing, e.g. in the VS code output panel (once rewritten to rust)

Reply:
how would you determine whether to show these errors as log messages, as opposed to other not separately handled `error!()` that we always want to show?

Reply:
I guess the main challenge is that we use `log` for both user-facing errors and tracing. I would recommend using the `log-tracing` for now. We never want to log any tracing events as user-visible messages.

Reply:
switching to tracing is easy, i just have to figure out how to get the currrent `warning`/`error` formatting there

Reply:
I'm not sure if we are on the same page. What I'm trying to say is that I don't think we should use tracing to show user facing errors. Actually, we should avoid that tracing uses `log` to output its messages because they then end up in the console output. 

I think my preference would be to instead use e.g. a rotating file logger to which we write our tracing logs. This would give us an additional debugging tool as well and prevents that the two log-streams from interleaving.

Reply:
No the logging works, i meant something simpler: `fern` gives us custom formatting for warnings and errors, for tracing we need to either switch to tracing logging style or implement our own layer that formats them in the `**warning**: {message}` style:
```
$ ruff --select PLR2 --no-cache asdf
warning: Failed to lint asdf: No such file or directory (os error 2)
$ target/debug/ruff --select PLR2 --no-cache asdf
2023-09-14T10:39:55.819547Z  WARN ruff_cli::commands::check: Failed to lint asdf: No such file or directory (os error 2)  
```

![image](https://github.com/astral-sh/ruff/assets/6826232/e8ac3041-cef5-4511-bde9-34c022b536b1)

I surprisingly couldn't find any example of how to intersect `warn!` and `error!` nor how to get our plain warn/error style without reimplementing the whole event formatting.

Reply:
But should tracing warnings even be user facing? I see tracing as a development tool and not a way to communicate to users.

Reply:
(I interpreted this discussion as non-blocking for the PR, but let me know if that's incorrect.)